src/Hosting/Hosting/src/Internal/StartupLoader.cs

```diff
@@ -303,7 +303,7 @@ internal static ConfigureServicesBuilder FindConfigureServicesDelegate([Dynamica
             return new ConfigureServicesBuilder(servicesMethod);
         }
 
-        private static MethodInfo FindMethod([DynamicallyAccessedMembers(StartupLinkerOptions.Accessibility)] Type startupType, string methodName, string environmentName, Type returnType = null, bool required = true)
+        private static MethodInfo? FindMethod([DynamicallyAccessedMembers(StartupLinkerOptions.Accessibility)] Type startupType, string methodName, string environmentName, Type? returnType = null, bool required = true)
```

Comment:
If you super feel like it, you could refactor this as `FindMethod` and `FindMethodRequired` and have the second overload guarantee non-nulless. It's a bit cleaner than varying the null-ness of the return value based on a parameter.

Reply:
I didn't because there is just one place where required is true.